static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,549 @@
+import React, { useRef, useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import { Link, useParams } from "react-router-dom";
+import { makeStyles } from "@material-ui/core/styles";
+
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import InputLabel from "@material-ui/core/InputLabel";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import FormControl from "@material-ui/core/FormControl";
+import Button from "@material-ui/core/Button";
+import Input from "@material-ui/core/Input";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import Typography from "@material-ui/core/Typography";
+import Box from "@material-ui/core/Box";
+import Dygraph from "dygraphs";
+
+import TextLoop from "react-text-loop";
+import { useForm, Controller } from "react-hook-form";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+
+import * as photometryActions from "../ducks/photometry";
+
+const useStyles = makeStyles((theme) => ({
+  copyb: {
+    display: "block",
+    border: "1px solid blue",
+  },
+  dygraphChart: {
+    position: "relative",
+    left: "1px",
+    right: "1px",
+    top: "1px",
+    bottom: "1px",
+    minWidth: "100%",
+    maxHeight: "100%",
+  },
+  media: {
+    maxWidth: "100%",
+    width: "95%",
+  },
+  button: {
+    margin: theme.spacing(1),
+  },
+  paper: {
+    width: "100%",
+    padding: theme.spacing(1),
+    textAlign: "left",
+    color: theme.palette.text.primary,
+  },
+  items: {
+    maxWidth: "100%",
+    width: "10rem",
+    fullWidth: "true",
+    display: "flex",
+    wrap: "nowrap",
+  },
+  spinner: {
+    position: "relative",
+    margin: "auto",
+    width: "50%",
+    fontWeight: "bold",
+    fontSize: "1.25rem",
+    textAlign: "center",
+  },
+  labels: {
+    fontSize: "0.75rem",
+  },
+}));
+
+function transpose(mat) {
+  // transpose matrix
+  return mat[0].map((col, i) => mat.map((row) => row[i]));
+}
+
+function plotdata(xx, yy, me) {
+  const dat = transpose([xx, yy])
+    .map((x, i) => [x[0], [x[1], me[i]]])
+    .sort((a, b) => b[0] - a[0]);
+  // eslint-disable-next-line no-new
+  new Dygraph(document.getElementById("dataplot"), dat, {
```

Comment:
document.getElementById should probably be replaced with a ref, as already done below for another element.

Reply:
Done.